### DISCUSSION TITLE
ESP8266 pins and multiple BMS

### DISCUSSION THREAD

**Discussion by andzie82** [syssi/esphome-jk-bms#223] [Q&A]
Hi, what pins should I use on my 8266, I saw the following on ESPHome,
 “The ESP8266 has two UARTs; the second of which is TX-only. Only a limited set of pins can be used. UART0 may use either tx_pin: GPIO1 and rx_pin: GPIO3, or tx_pin: GPIO15 and rx_pin: GPIO13. UART1 must use tx_pin: GPIO2. Any other combination of pins will result in use of a software UART.” 
 I guess that means that I only can connect one BMS per 8266 and it should be tx_pin: GPIO15 and rx_pin: GPIO13?

![4DE7D5B5-155F-4EBD-8F6D-8BA89C6E7E8E](https://user-images.githubusercontent.com/48380052/207946581-8339d57d-86f3-4a8f-a16b-6a323896e0f8.jpeg)

**Comment by syssi** [comment]
If you use another pair of GPIOs a so called software serial implementation is used instead of the hardware UART interface. GPIO13 and GPIO15 are a good choice if you don't want to use GPIO1 and GPIO3.

A ESP32 has 3 hardware UARTs accessible via any GPIO because of a routing matrix.
